File: Leetcode/Matrix/73.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        zeroes = []
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    zeroes.append((i, j))

        for zero in zeroes:
            x, y = zero
            for k in range(len(matrix[0])):
                matrix[x][k] = 0
            for k in range(len(matrix)):
                matrix[k][y] = 0
                

File: Leetcode/Matrix/73.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function (matrix) {
  let m = matrix.length;
  let n = matrix[0].length;

  function setZeros(x, y) {
    for (let i = 0; i < m; i++) matrix[i][y] = 0;
    for (let i = 0; i < n; i++) matrix[x][i] = 0;
  }

  let zeros = [];
  for (let i = 0; i < m; i++) {
    for (let j = 0; j < n; j++) {
      if (matrix[i][j] === 0) zeros.push([i, j]);
    }
  }

  for (let i = 0; i < zeros.length; i++) {
    let [x, y] = zeros[i];
    setZeros(x, y);
  }

  return matrix;
};
